Sources/SwiftBSON/BSONDocument.swift

```diff
@@ -519,40 +519,35 @@ extension BSONDocument: CustomStringConvertible {
 }
 
 extension BSONDocument {
-    /// Helper function to check for BSON document equality regardless of the order of key-value pairs
-    internal func equalsIgnoreOrderHelper(a: BSONDocument, b: BSONDocument) -> Bool {
-        let keysA = a.keys.sorted()
-        let keysB = b.keys.sorted()
-
-        guard keysA == keysB else {
+    /**
+     * Returns whether this `BSONDocument` contains exactly the same key/value pairs as the provided `BSONDocument`,
+     * regardless of the order of the keys.
+     *
+     * Warning: This method is much less efficient than checking for regular equality since the document is internally
+     * ordered.
+     *
+     * - Parameters:
+     *   - other: a `BSONDocument` to compare this document with.
+     *
+     * - Returns: a `Bool` indicating whether the two documents are equal.
+     */
+    public func equalsIgnoreKeyOrder(_ other: BSONDocument) -> Bool {
+        guard self.keys.count == other.keys.count else {
```

Comment:
`BSONDocument.keys` returns a `[String]` containing all the keys in the document in order, meaning it requires a traversal of the entire document (i.e. O(n)). `BSONDocument.count` instead uses the cached `keySet`, so no traversal is required (i.e. O(1)).